workers/unity/Packages/com.improbable.gdk.querybasedinteresthelper/Constraint.cs

```diff
@@ -0,0 +1,390 @@
+        /// <remarks>
+        ///     This constraint defines a box relative to the position of the entity.
+        /// </remarks>
+        /// <returns>
+        ///     A Constraint object.
+        /// </returns>
+        public static Constraint RelativeBox(double xWidth, double yHeight, double zDepth)
+        {
+            var constraint = Default();
+            constraint.RelativeBoxConstraint = new ComponentInterest.RelativeBoxConstraint
+            {
+                EdgeLength = new EdgeLength(xWidth, yHeight, zDepth)
+            };
+            return new Constraint(constraint);
+        }
+
+        /// <summary>
+        ///     Creates a Constraint object with an EntityId constraint.
+        /// </summary>
+        /// <param name="entityId">
+        ///     EntityId of the .
+        /// </param>
+        /// <returns>
+        ///     A Constraint object.
+        /// </returns>
+        public static Constraint EntityId(EntityId entityId)
+        {
+            var constraint = Default();
+            constraint.EntityIdConstraint = entityId.Id;
+            return new Constraint(constraint);
+        }
+
+        /// <summary>
+        ///     Creates a Constraint object with an EntityId constraint.
+        /// </summary>
+        /// <typeparam name="T">
+        ///     Type of the component to constrain.
+        /// </typeparam>
+        /// <returns>
+        ///     A Constraint object.
+        /// </returns>
+        public static Constraint Component<T>() where T : ISpatialComponentData
+        {
+            var constraint = Default();
+            constraint.ComponentConstraint = Dynamic.GetComponentId<T>();
+            return new Constraint(constraint);
+        }
+
+        /// <summary>
+        ///     Creates a Constraint object with a Component constraint.
+        /// </summary>
+        /// <param name="componentId">
+        ///     ID of the component to constrain.
+        /// </param>
+        /// <returns>
+        ///     A Constraint object.
+        /// </returns>
+        public static Constraint Component(uint componentId)
+        {
+            var constraint = Default();
+            constraint.ComponentConstraint = componentId;
+            return new Constraint(constraint);
+        }
+
+        /// <summary>
+        ///     Creates a Constraint object with an And constraint.
+        /// </summary>
+        /// <param name="constraint">
+        ///     First constraint in the list of conjunctions.
+        /// </param>
+        /// <param name="constraints">
+        ///     Further constraints for the list of conjunctions.
+        /// </param>
+        /// <remarks>
+        ///     At least one constraint must be provided to create an "All" QueryConstraint.
+        /// </remarks>
+        /// <returns>
+        ///     A Constraint object.
+        /// </returns>
+        public static Constraint All(Constraint constraint, params Constraint[] constraints)
```

Comment:
I'd suggest adding an overload for Any/All that takes an `IEnumerable<Constraint>` - I remember Klang giving feedback about this on the SDK version of this.
You lose the compile-time guarantee that you have to provide at least one constraint, but it does allow for more "native" use cases (say, constructing from a static List<> of constraints)